### PR TITLE
Upgrade storybook to v7

### PR DESCRIPTION
**Pull request by micheal-parks**
This PR upgrades us to Storybook v7. There are a lot of good changes here, and the [blog post illustrates them best](https://storybook.js.org/blog/7-0-beta/). Storybook 7 is in beta, but considering that it's a dev dependency and the issues that we're having with v6 (in particular resolving peer dependencies) that have been fixed, I think it's worth it to upgrade now.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "@viamrobotics/prime",
   "version": "0.0.100",
   "license": "Apache-2.0",
-  "type": "commonjs",
+  "type": "module",
```
Comment (micheal-parks) [inline package.json:5]: The latest version of the vite plugin for svelte no longer supports commonJS, and luckily storybook 7 has no issues with using ES modules (unlike v6).

File: package.json [hunk 2]
```diff
@@ -26,50 +26,50 @@
...
-    "svelte": "^3.52.0",
-    "svelte-check": "^2.9.2",
-    "svelte-loader": "^3.1.3",
-    "svelte-preprocess": "^4.10.7",
-    "tailwindcss": "^3.2.1",
-    "terser": "^5.15.1",
+    "postcss": "^8.4.20",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "storybook": "^7.0.0-beta.13",
```
Comment (micheal-parks) [inline package.json:65]: This is only made obvious by the `package-lock.json`, but dependencies have been substantially reduced.

File: src/elements/select/multiselect.svelte [hunk 3]
```diff
@@ -116,7 +116,7 @@ const handleEnter = () => {
 
 const handleChange = (changedOption: string) => {
   if (parsedSelected.includes(changedOption)) {
-    const newValue = [...parsedSelected.filter(item => item !== changedOption)];
+    const newValue = parsedSelected.filter(item => item !== changedOption);
```
Comment (micheal-parks) [inline src/elements/select/multiselect.svelte:119]: Most of the changes in `src` are new linting rules. For example here: filter copies arrays so ESLint removed this unnecessary spread.

File: src/stories/docs/contributing.stories.mdx [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import { Meta } from '@storybook/addon-docs';
```
Comment (micheal-parks) [inline src/stories/docs/contributing.stories.mdx:1]: These need to be explicit imports in mdx2

**Review by mrloureed — APPROVED** [review]
Works and LGTM.

Any chance we can remove the overflow hidden for docs?
image

**Review by michaellee1019 — APPROVED** [review]

**Pull request merged**